pkg/rtc/mediatrackreceiver.go

```diff
@@ -161,19 +161,21 @@ func (t *MediaTrackReceiver) SetupReceiver(receiver sfu.TrackReceiver, priority
 	receivers := slices.Clone(t.receivers)
 
 	// codec position maybe taken by DummyReceiver, check and upgrade to WebRTCReceiver
-	var existingReceiver bool
-	for _, r := range receivers {
+	idx := -1
+	for i, r := range receivers {
 		if strings.EqualFold(r.Codec().MimeType, receiver.Codec().MimeType) {
-			existingReceiver = true
-			if d, ok := r.TrackReceiver.(*DummyReceiver); ok {
-				d.Upgrade(receiver)
-			}
+			idx = i
 			break
 		}
 	}
-	if !existingReceiver {
-		receivers = append(receivers, &simulcastReceiver{TrackReceiver: receiver, priority: priority})
+	if idx != -1 {
+		if d, ok := receivers[idx].TrackReceiver.(*DummyReceiver); ok {
+			d.Upgrade(receiver)
+		}
+		// replace receiver
+		receivers = slices.Delete(receivers, idx, idx+1)
 	}
+	receivers = append(receivers, &simulcastReceiver{TrackReceiver: receiver, priority: priority})
```

Comment:
It would be better to keep the idx instead of pushing it to the back as the relay node expects the codec order unchanged during resolve.

Reply:
There is also a sort by priority just after that code. So, the ordering will be by priority irrespective of the order before that. Do you think that is not enough?

Reply:
It is ok, I missed it:)

Reply:
Cool, thank you :-)